lib/Service/ConfigService.php

```diff
@@ -0,0 +1,276 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OCA\Gitlab\Service;
+
+use OCA\Gitlab\AppInfo\Application;
+use OCP\IConfig;
+use OCP\PreConditionNotMetException;
+
+class ConfigService {
+	public function __construct(
+		private IConfig $config,
+	) {
+	}
+
+	public function getAdminClientId(): string {
+		return $this->config->getAppValue(Application::APP_ID, 'client_id');
+	}
+
+	public function setAdminClientId(string $clientId): void {
+		$this->config->setAppValue(Application::APP_ID, 'client_id', $clientId);
+	}
+
+	public function hasAdminClientSecret(): bool {
+		return $this->getAdminClientSecret() !== '';
+	}
+
+	public function getAdminClientSecret(): string {
+		return $this->config->getAppValue(Application::APP_ID, 'client_secret');
+	}
+
+	public function setAdminClientSecret(string $clientSecret): void {
+		$this->config->setAppValue(Application::APP_ID, 'client_secret', $clientSecret);
+	}
+
+	public function getAdminOauthUrl(): string {
+		return $this->config->getAppValue(Application::APP_ID, 'oauth_instance_url', Application::DEFAULT_GITLAB_URL);
+	}
+
+	public function setAdminOauthUrl(string $url): void {
+		$this->config->setAppValue(Application::APP_ID, 'oauth_instance_url', $url);
+	}
```

Comment:
let's add `?: Application::DEFAULT_GITLAB_URL` to the get since the admin oauth url can be set to an empty string. Or you might prefer to do it in the set function.